qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -469,8 +545,26 @@ def upload_curve(
 
         sensor_unit = Model_325_Curve.validate_datadict(data_dict)
 
-        curve = self.curve[index]
+        curve = self.curve[index - 1]
         curve.curve_name(name)
         curve.serial_number(serial_number)
         curve.format(f"{sensor_unit}/K")
         curve.set_data(data_dict, sensor_unit=sensor_unit)
+
+    def upload_curve_from_file(self, index: int, file_path: str) -> None:
+        """
+        Upload a curve from a curve file. Note that we only support
+        curve files with extension *.330
+        """
+        _, filename = os.path.split(file_path)
```

Comment:
why is this needed? isn't `filename.endswith(".330") == file_path.endswith(".330")`?